test/testdefs.jl

```diff
@@ -4,10 +4,17 @@ using Base.Test
 
 function runtests(name)
     @printf("     \033[1m*\033[0m \033[31m%-21s\033[0m", name)
-    tt = @elapsed include("$name.jl")
+    coolness = @timed @testset "$name" begin
+        include("$name.jl")
+    end
     rss = Sys.maxrss()
-    @printf(" in %6.2f seconds, maxrss %7.2f MB\n", tt, rss / 2^20)
-    rss
+    @printf(" maxrss %7.2f MB\n", rss / 2^20)
+    #coolness[1] is the testset
+    passes,fails,error,broken,c_passes,c_fails,c_errors,c_broken = Base.Test.get_test_counts(coolness[1])
+    if coolness[1].anynonpass == false
+        coolness = ((passes+c_passes,broken+c_broken), coolness[2], coolness[3], coolness[4], coolness[5])
+    end
+    vcat([coolness...], rss)
```

Comment:
Might be. I should probably change the name of that variable to something more descriptive as well.